velox/common/memory/Memory.h

```diff
@@ -67,87 +67,90 @@ constexpr uint16_t kDefaultAlignment = 64;
 
 class ScopedMemoryPool;
 
-class MemoryPool {
+class MemoryPool : public std::enable_shared_from_this<MemoryPool> {
  public:
-  virtual ~MemoryPool() {}
+  MemoryPool(const std::string& name, std::shared_ptr<MemoryPool> parent);
+  virtual ~MemoryPool();
+
+  ////////////////////////// tree methods ////////////////////////////
+  virtual const std::string& getName() const;
```

Comment:
- getName() -> name()
- getChildCount() -> size()

Reply:
Rename getName to name. I might keep getChildCount as seems to me more clear than size() in name.

Reply:
OK. Can you then rename to childCount() ?

Reply:
or numChildren()?